MailerModule: Logging (#4285)
vivid-planet/comet: docs/docs/3-features-modules/11-mailer-module/index.md

async publishAllProducts(): Promise<boolean> {
    ...
    await this.mailerService.sendMail({
        mailTypeForLogging: "products-published",
        to: "product-manager@comet-dxp.com",
        cc: "vice-product-manager@comet-dxp.com",
        subject: "All products have been published",
    });
    ...
}
```

---

## Deleting Old Mail Logs

This is done automatically after creating a new mail log entry. For any custom cleanup you should use the MailLog Repository in your application.

---

## Mail Log Creation and Updates

Mail logs are created automatically whenever an email is sent using the `sendMail` method of the `MailerService`. Each log entry records details such as recipients, subject, mail options, additional data, and the mail type for filtering and statistics. The log is created before the email is sent.

After the email is sent, the log entry is updated with the result of the send operation (such as the message ID and status). If mail logging is disabled via configuration, no log entries are created or updated.

> Mail logs are not created when using the `logMail` option of `sendMail`.

Reviewer: ```suggestion
Mail logs can be disabled using the `logMail` option of `sendMail`.
```
